cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -246,7 +248,8 @@ private void resetRegistry() {
     registry = registryFactory.create();
     for (Map.Entry<String, ? extends DatasetModule> entry : defaultModules.entrySet()) {
       LOG.info("Adding Default module: " + entry.getKey() + " " + this.toString());
-      add(entry.getKey(), entry.getValue(), true);
+      // default modules in system namespace
+      add(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, entry.getKey()), entry.getValue(), true);
```

Comment:
I'm a little confused, don't you currently have to construct the Id everywhere because the map only contains the name?  The intent behind the comment is to only hardcode the system namespace in one place instead of everywhere the defaults are used.  If that's not the case, then ignore the comment.